android/src/main/java/com/swmansion/reanimated/layoutReanimation/SharedTransitionManager.java

```diff
@@ -0,0 +1,400 @@
+      Snapshot targetViewSnapshot = mSnapshotRegistry.get(viewTarget.getId());
+
+      if (!mCurrentSharedTransitionViews.containsKey(viewSource.getId())) {
+        mCurrentSharedTransitionViews.put(viewSource.getId(), viewSource);
+      }
+      if (!mCurrentSharedTransitionViews.containsKey(viewTarget.getId())) {
+        mCurrentSharedTransitionViews.put(viewTarget.getId(), viewTarget);
+      }
+      SharedElement sharedElement =
+          new SharedElement(viewSource, sourceViewSnapshot, viewTarget, targetViewSnapshot);
+      sharedElements.add(sharedElement);
+    }
+    return sharedElements;
+  }
+
+  private void setupTransitionContainer() {
+    if (!mIsSharedTransitionActive) {
+      mIsSharedTransitionActive = true;
+      ReactContext context = mAnimationsManager.getContext();
+      Activity currentActivity = context.getCurrentActivity();
+      if (currentActivity == null) {
+        return;
+      }
+      ViewGroup rootView = (ViewGroup) currentActivity.getWindow().getDecorView().getRootView();
+      if (mTransitionContainer == null) {
+        mTransitionContainer = new ReactViewGroup(context);
+      }
+      rootView.addView(mTransitionContainer);
+      mTransitionContainer.bringToFront();
+    }
+  }
+
+  private void reparentSharedViewsForCurrentTransition(List<SharedElement> sharedElements) {
+    for (SharedElement sharedElement : sharedElements) {
+      View viewSource = sharedElement.sourceView;
+      View viewTarget = sharedElement.targetView;
+
+      if (!mSharedTransitionParent.containsKey(viewSource.getId())) {
+        mSharedTransitionParent.put(viewSource.getId(), (View) viewSource.getParent());
+        mSharedTransitionInParentIndex.put(
+            viewSource.getId(), ((ViewGroup) viewSource.getParent()).indexOfChild(viewSource));
+        ((ViewGroup) viewSource.getParent()).removeView(viewSource);
+        ((ViewGroup) mTransitionContainer).addView(viewSource);
+      }
+
+      if (!mSharedTransitionParent.containsKey(viewTarget.getId())) {
+        mSharedTransitionParent.put(viewTarget.getId(), (View) viewTarget.getParent());
+        mSharedTransitionInParentIndex.put(
+            viewTarget.getId(), ((ViewGroup) viewTarget.getParent()).indexOfChild(viewTarget));
+        ((ViewGroup) viewTarget.getParent()).removeView(viewTarget);
+        ((ViewGroup) mTransitionContainer).addView(viewTarget);
+      }
+    }
+  }
+
+  private void startSharedTransition(List<SharedElement> sharedElements) {
+    for (SharedElement sharedElement : sharedElements) {
+      onViewTransition(
+          sharedElement.sourceView,
+          sharedElement.sourceViewSnapshot,
+          sharedElement.targetViewSnapshot);
+      onViewTransition(
+          sharedElement.targetView,
+          sharedElement.sourceViewSnapshot,
+          sharedElement.targetViewSnapshot);
+    }
+  }
+
+  private void onViewTransition(View view, Snapshot before, Snapshot after) {
```

Comment:
```suggestion
  private void startViewTransition(View view, Snapshot before, Snapshot after) {
```
might me a better name